opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java

```diff
@@ -0,0 +1,251 @@
+        Mockito.doReturn(database).when(repository).getReadableDatabase();
+        MatrixCursor matrixCursor = new MatrixCursor(new String[]{
+                "location_id"});
+        Mockito.doReturn(matrixCursor).when(database).rawQuery(Mockito.any(), Mockito.any());
+        List<String> locationList = ReportDao.extractRecordedLocations();
+        Mockito.verify(database).rawQuery(Mockito.anyString(), Mockito.any());
+        Assert.assertEquals(locationList.size(), 0);
+    }
+
+    @Test
+    public void testEligibleChildrenReport() {
+        Mockito.doReturn(database).when(repository).getReadableDatabase();
+        MatrixCursor matrixCursor = new MatrixCursor(new String[]{
+                "base_entity_id"});
+        matrixCursor.addRow(new Object[]{"abdb6140-d54b-49d4-89e7-6f96c839c62b"});
+        Mockito.doReturn(matrixCursor).when(database).rawQuery(Mockito.any(), Mockito.any());
+        String communityId = "d5ff0ea1-bbc5-424d-84c2-5b084e10ef90";
+        Date dueDate = new DateTime().plusDays(7).toDate();
+        List<EligibleChild> eligibleChildren = ReportDao.eligibleChildrenReport(communityId, dueDate);
+        Mockito.verify(database).rawQuery(Mockito.anyString(), Mockito.any());
+        Assert.assertEquals(eligibleChildren.get(0).getID(), "abdb6140-d54b-49d4-89e7-6f96c839c62b");
+    }
+
+    @Test
+    public void testEligibleChildrenReportReturnsEmptyList() {
+        Mockito.doReturn(database).when(repository).getReadableDatabase();
+        MatrixCursor matrixCursor = new MatrixCursor(new String[]{
+                "base_entity_id"});
+        Mockito.doReturn(matrixCursor).when(database).rawQuery(Mockito.any(), Mockito.any());
+        String communityId = "d5ff0ea1-bbc5-424d-84c2-5b084e10ef90";
+        Date dueDate = new DateTime().plusDays(7).toDate();
+        List<EligibleChild> eligibleChildren = ReportDao.eligibleChildrenReport(communityId, dueDate);
+        Mockito.verify(database).rawQuery(Mockito.anyString(), Mockito.any());
+        Assert.assertEquals(eligibleChildren.size(), 0);
+    }
+
+    @Test
+    public void testVillageDosesReport() {
+        Mockito.doReturn(database).when(repository).getReadableDatabase();
+        MatrixCursor matrixCursor = new MatrixCursor(new String[]{
+                "scheduleName"});
+        matrixCursor.addRow(new Object[]{"ROTA 1"});
+        Mockito.doReturn(matrixCursor).when(database).rawQuery(Mockito.any(), Mockito.any());
+        String villageName = "Grenaligne";
+        String communityId = "d5ff0ea1-bbc5-424d-84c2-5b084e10ef90";
+        Date dueDate = new DateTime().plusDays(7).toDate();
+        Map<String, Integer> map = new TreeMap<>();
+        map.put("ROTA", 0);
+        List<VillageDose> villageDoseList = ReportDao.villageDosesReport(villageName,communityId, dueDate);
+        Mockito.verify(database).rawQuery(Mockito.anyString(), Mockito.any());
+        Assert.assertEquals(villageDoseList.get(0).getRecurringServices(), map);
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.*'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)